src/main/java/io/jenkins/plugins/fontawesome/FontAwesomeIcons.java

```diff
@@ -0,0 +1,110 @@
+package io.jenkins.plugins.fontawesome;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.*;
+import java.util.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Stream;
+
+/**
+ * Utility to work with icons provided by the font-awesome-api-plugin.
+ *
+ * @author strangelookingnerd
+ */
+public final class FontAwesomeIcons {
+
+    private static final Logger LOGGER = Logger.getLogger(FontAwesomeIcons.class.getName());
+    private static final String SVG_FILE_ENDING = ".svg";
+    private static final String IMAGES_SYMBOLS_PATH = "images/symbols/";
+    private static final String FONT_AWESOME_API_PLUGIN = "font-awesome-api";
+    private static final String ICON_CLASS_NAME_PATTERN = "symbol-%s plugin-" + FONT_AWESOME_API_PLUGIN;
+
+    private FontAwesomeIcons() {
+        // hidden
+    }
+
+    private static Map<String, String> getIconsFromClasspath(final SvgTag.FontAwesomeStyle filter) {
+        try {
+            Enumeration<URL> urls = FontAwesomeIcons.class.getClassLoader().getResources(IMAGES_SYMBOLS_PATH);
+
+            while (urls.hasMoreElements()) {
+                URL url = urls.nextElement();
+
+                if (StringUtils.contains(url.toExternalForm(), FONT_AWESOME_API_PLUGIN)) {
+                    URI uri = url.toURI();
+
+                    if (StringUtils.equals(uri.getScheme(), "jar")) {
+                        try (FileSystem fileSystem = FileSystems.newFileSystem(uri, Collections.emptyMap())) {
+                            return collectIcons(fileSystem.getPath(IMAGES_SYMBOLS_PATH), filter);
+                        }
+                    }
+                    else {
+                        return collectIcons(Paths.get(uri), filter);
+                    }
+                }
+            }
+        }
+        catch (IOException | URISyntaxException ex) {
+            LOGGER.log(Level.WARNING, "Unable to read available icons: Resource unavailable.", ex);
+        }
+
+        return new LinkedHashMap<>();
+    }
+
+    private static Map<String, String> collectIcons(final Path path, final SvgTag.FontAwesomeStyle filter) throws IOException {
+        Map<String, String> icons = new LinkedHashMap<>();
+
+        if (path != null) {
```

Comment:
Shouldn't this better throw an exception?

Reply:
I think it is very unlikely that an exception would be thrown here in the first place. A caller could not do anything useful with the exception either.